src/main/java/frc/robot/Drive.java

```diff
@@ -0,0 +1,212 @@
+        System.out.println("Error: " + error);
+        //error between desiredAngle and our current angle is established
+        if(error < -2 || error > 2) {
+          //While loop will continue as long as error is not inbetween the slack range of -3 to 3
+          error = desiredAngle - gyro.getAngle();
+          //This calculates at the start of every loop to determine which way the robot will turn
+          if(error > 0) {
+              //If error is positive, the robot turns right
+              leftFront.set(speed);
+              rightFront.set(-speed);
+              leftBack.set(speed);
+              rightBack.set(-speed);
+          }
+          else {
+              //If error is negative, the robot turns left
+              leftFront.set(-speed);
+              rightFront.set(speed);
+              leftBack.set(-speed);
+              rightBack.set(speed);
+          }
+        }
+        else {
+          leftFront.stopMotor();
+          rightFront.stopMotor();
+          leftBack.stopMotor();
+          rightBack.stopMotor();
+          turnDrive = true;
+        }
+
+        //Stop all motors
+    }
+
+    public void angleAlign() {
+      double speed = 0;
+      double k = 0.03;
+      double angleOffset = aprilTags.getHorizontalDegToTarget();
+      alignDone = false;
+
+      if (angleOffset < -0.5 || angleOffset > 0.5) {
+        speed = angleOffset * k;
+        if (Math.abs(speed) > 0.1) {
+          speed = Math.copySign(0.1, speed);
+        }
+        else if (Math.abs(speed) < 0.075) {
+          speed = Math.copySign(0.075, speed);
+        }
+        leftFront.set(speed);
+        rightFront.set(-speed);
+        leftBack.set(speed);
+        rightBack.set(-speed);
+      }
+      else {
+        leftFront.stopMotor();
+        rightFront.stopMotor();
+        leftBack.stopMotor();
+        rightBack.stopMotor();
+        alignDone = true;
+      }
+    }
+
+    // public void driveToAprilTag(String direction, double timeOut) {
+    //   double speed = 0.20;
+    //   double distance = aprilTags.estimateHorizontalDistance() - 45;
+    //   double timeStarted = 
+
+    //   if (distance > 1) {
+    //     leftFront.set(speed);
+    //     rightFront.set(speed);
+    //     leftBack.set(speed);
+    //     rightBack.set(speed);  
+    //   }
+    //   else if (distance < 1) {
+    //     leftFront.set(-speed);
+    //     rightFront.set(-speed);
+    //     leftBack.set(-speed);
+    //     rightBack.set(-speed);
+    //   }
+      
+    //   if(Math.abs(lFrontEncoder.getPosition()) >= motorTarget || Math.abs(rFrontEncoder.getPosition()) >= motorTarget || Math.abs(lBackEncoder.getPosition()) >= motorTarget || Math.abs(rBackEncoder.getPosition()) >= motorTarget || System.currentTimeMillis() > (timeStarted + timeOut)) {
+    //     leftFront.set(0);
+    //     rightFront.set(0);
+    //     leftBack.set(0);
+    //     rightBack.set(0);
+    //     eDriveDone = true;
+    //   }
+  
+    // }
```

Comment:
I think this method might have the same functionality as distanceAlign() from the Align.java method